### PR TITLE
Do not validate licenseFile and licenseText values to prevent startup issues

### PR DESCRIPTION
**Pull request by ramonsmits**
Do not validate licenseFile and licenseText values to prevent startup issues due to invalid values

Resolves:

- https://github.com/Particular/NServiceBus/issues/7060

> No technical limitations are enforced at runtime when either no license is found or a license has expired.

Source: https://docs.particular.net/nservicebus/licensing/#throughput-limitations

### CODE REVIEW

**Review by SeanFeldman — COMMENTED** [review]
This may not be the right approach.
Let's assume I'm unintentionally providing content that is considered an invalid license, thinking it's right. How will I find out my license content differs from what I think it is?
I'd appreciate it if you could have a warning log.

**Review by johnsimons — CHANGES_REQUESTED** [review]
@ramonsmits in the [issue](https://github.com/Particular/NServiceBus/issues/7060) referenced, it seems the user is explicitly calling the API, which in this case is throwing an exception.
If the user wants to handle a scenario where the license string may be empty or null, then they have to handle that themselves before calling the API.

In other words, I think the validation error is correct and should not be removed.

**Comment by jpalac** [comment]
It feels like if you're specifically using `endpointConfiguration.LicensePath("PathToLicense")` then you intend to provide a path to a license file. If you do not have a path, then do not make the API call.
If a path is provided but it is invalid (doesn't exist) or the license found at the path is invalid or expired, then the code works as expected.

**Comment by ramonsmits** [comment]
The documentation states:

> No technical limitations are enforced at runtime when either no license is found or a license has expired.

However, this API is an explicit call. It is expected that *some* data to process is present.

File: src/NServiceBus.Core/Licensing/ConfigureLicenseExtensions.cs [hunk 1]
```diff
@@ -30,9 +27,7 @@ public static void License(this EndpointConfiguration config, string licenseText
     /// <remarks>The license provided via code-first API takes precedence over other license sources.</remarks>
     public static void LicensePath(this EndpointConfiguration config, string licenseFile)
     {
-        ArgumentNullException.ThrowIfNull(config);
-        ArgumentException.ThrowIfNullOrWhiteSpace(licenseFile);
-
+        // Intentionally not doing validation as passed value could be dynamic and (temporarily) be invalid
```
Comment (ramonsmits) [inline src/NServiceBus.Core/Licensing/ConfigureLicenseExtensions.cs:30]: What is strange here is that currently we do not do a `File.Exist`. If we defer that we could also defer ANY validation. However, now including a `File.Exist` is a breaking change.

Comment (johnsimons) [reply src/NServiceBus.Core/Licensing/ConfigureLicenseExtensions.cs:30]: It is a tricky situation, it may be nice to involve others to see their thoughts

**Comment by ramonsmits** [comment]
I'm not satisfied with these changes. Too big diff for what we want.

Alternative is to just add validation in the extension methods and use 

    LogManager.GetLogger<LicenseManager>().Error("LicensePath argument is invalid and will be ignored")

Problem with that is that although no ArgumentException is raised the behavior will be different as now we will scan a bunch of default locations for the presence of a license file.

**Pull request closed without merging**